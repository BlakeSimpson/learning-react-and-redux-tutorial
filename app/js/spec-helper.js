import 'raf/polyfill';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

export const pointsMock = [
  {
    id: 1,
    x: 450,
    y: 110,
    details: {
      name: 'The Wall',
      house: "Night's Watch",
      words: 'Night gathers, and now my watch begins.'
    },
    favourite: true
  },
  {
    id: 2,
    x: 375,
    y: 355,
    details: {
      name: 'Winterfell',
      house: 'Stark',
      words: 'Winter is Coming'
    },
    favourite: false
  }
];
