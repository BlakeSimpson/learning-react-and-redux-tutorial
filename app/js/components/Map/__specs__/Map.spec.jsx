import React from 'react';
import { shallow } from 'enzyme';

import { pointsMock } from '../../../spec-helper';
import { Map } from '../';

describe('Map component', () => {
  describe('when there are no points', () => {
    it('matches the snapshot', () => {
      const wrapper = shallow(<Map points={[]} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when there are points', () => {
    it('matches the snapshot', () => {
      const wrapper = shallow(<Map points={pointsMock} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
