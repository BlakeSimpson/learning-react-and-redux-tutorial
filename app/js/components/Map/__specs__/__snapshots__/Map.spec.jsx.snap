// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component when there are no points matches the snapshot 1`] = `<div />`;

exports[`Map component when there are points matches the snapshot 1`] = `
<div>
  <Connect(Pointer)
    details={
      Object {
        "house": "Night's Watch",
        "name": "The Wall",
        "words": "Night gathers, and now my watch begins.",
      }
    }
    favourite={true}
    id="point-1"
    key="point-1"
    x={450}
    y={110}
  />
  <Connect(Pointer)
    details={
      Object {
        "house": "Stark",
        "name": "Winterfell",
        "words": "Winter is Coming",
      }
    }
    favourite={false}
    id="point-2"
    key="point-2"
    x={375}
    y={355}
  />
</div>
`;
